### ISSUE TITLE
DNS record deletion causes exception

### ISSUE THREAD

**Issue by nitr0man** [cloudflare/python-cloudflare#92]
DNS record deletion causes exception - it seems like API returns error, but without 'code' (it's not mandatory)

_raw() call returns  `{'error': 'You cannot use this API for domains with a .cf, .ga, .gq, .ml, or .tk TLD (top-level domain). To configure the DNS settings for this domain, use the Cloudflare Dashboard.'}`

Exception trace: 

```
In [10]: record = cf.zones.dns_records.get(zone['id'])[0]                                                                                                                                                                                     

In [11]: record['zone_id'], record['id']                                                                                                                                                                                                      
Out[11]: ('9d2e3522d6f813a1d32c85d0fe137443', 'a47180b9c7daa2e8ec92fcc7a5fb2da8')

In [12]: zone = cf.zones.get(params={'name': '42cc-testing-site.tk'})[0]                                                                                                                                                                      

In [13]: record = cf.zones.dns_records.get(zone['id'])[0]                                                                                                                                                                                     

In [14]: record['zone_id'], record['id']                                                                                                                                                                                                      
Out[14]: ('9d2e3522d6f813a1d32c85d0fe137443', 'a47180b9c7daa2e8ec92fcc7a5fb2da8')

In [15]: cf.zones.dns_records.delete(record['zone_id'], record['id'])                                                                                                                                                                         
---------------------------------------------------------------------------
KeyError                                  Traceback (most recent call last)
<ipython-input-15-06326d1a5ea4> in <module>
----> 1 cf.zones.dns_records.delete(record['zone_id'], record['id'])

~/.virtualenvs/pbn/lib/python3.6/site-packages/CloudFlare/cloudflare.py in delete(self, identifier1, identifier2, identifier3, params, data)
    683             return self._base.call_with_auth('DELETE', self._parts,
    684                                              identifier1, identifier2, identifier3,
--> 685                                              params, data)
    686 
    687     class _AddWithAuthUnwrapped(object):

~/.virtualenvs/pbn/lib/python3.6/site-packages/CloudFlare/cloudflare.py in call_with_auth(self, method, parts, identifier1, identifier2, identifier3, params, data, files)
    121             return self._call(method, headers, parts,
    122                               identifier1, identifier2, identifier3,
--> 123                               params, data, files)
    124 
    125         def call_with_auth_unwrapped(self, method, parts,

~/.virtualenvs/pbn/lib/python3.6/site-packages/CloudFlare/cloudflare.py in _call(self, method, headers, parts, identifier1, identifier2, identifier3, params, data, files)
    486             if response_data['success'] is False:
    487                 errors = response_data['errors'][0]
--> 488                 code = errors['code']
    489                 if 'message' in errors:
    490                     message = errors['message']

KeyError: 'code'

```

**Comment by ddzzhen** [comment]
OMG , have to buy one for api then .

**Comment by mahtin** [comment]
This is fixed in `2.7.1` release. The error you are getting is correct; however, I'm not sure why that specific API call doesn't return a `code` value. I could not replicate it and yet I could see exactly how to fix it. Please retest.

**Comment by nitr0man** [comment]
Ok I'll test it.
It's easy to reproduce: just create .tk zone, and try to add and then - delete record.
